### PR TITLE
fix(frontend): improper use of removeEventListener, calls to scroll e…

### PR DESCRIPTION
**Pull request by EugeneHlushko**
…vent callback without throttling.

* Dont try to remove non existent listener, it must be the same function or reference as one that was bind before. E.g. when component unmounts `_recalculate`'s function instance returned by `.bind` will still be called on each scroll without this fix.
* Use throttle to not overload browser

Just to be clear .bind() returns a new function each time.

### CODE REVIEW

**Comment by skipjack** [comment]
Interesting, hadn't heard of `throttle` before and you're definitely correct re `.bind()` creating new function. I'll leave this to the new maintainers to test and merge.

**Comment by EugeneHlushko** [comment]
Who is going to maintain after you go @skipjack ? Should i ping someone in this PR ?

**Comment by skipjack** [comment]
Yeah ping @jereminichelli @montogeek or @thedutchcoder. Whoever reviews these kinds of changes should make sure to pull the branch and test locally.

**Comment by montogeek** [comment]
I will take a look later

File: src/components/Sidebar/Sidebar.jsx [hunk 1]
```diff
@@ -48,7 +49,7 @@ export default class Sidebar extends Component {
                   title={ page.title }
                   anchors={ page.anchors }
                   currentPage= { currentPage }
```
Comment (dhruvdutt) [inline src/components/Sidebar/Sidebar.jsx]: Can you please clean extra whitespace here? ~`= {`~ => `={`

Comment (EugeneHlushko) [reply src/components/Sidebar/Sidebar.jsx]: guess it can be automated later on with prettier

**Comment by EugeneHlushko** [comment]
@montogeek any luck reviewing this pr ?

File: src/components/Sidebar/Sidebar.jsx [hunk 2]
```diff
@@ -60,28 +61,29 @@ export default class Sidebar extends Component {
...
     );
   }
 
   /**
    * Re-calculate fixed state and position
    *
    */
-  _recalculate() {
+  _recalculate = throttle(() => {
+    console.log('recalc running');
```
Comment (dhruvdutt) [inline src/components/Sidebar/Sidebar.jsx]: Please, clean `console` statements.

Comment (EugeneHlushko) [reply src/components/Sidebar/Sidebar.jsx]: LOL how did that get through 💃

Comment (EugeneHlushko) [reply src/components/Sidebar/Sidebar.jsx]: thanks mate, very good catch

Comment (dhruvdutt) [reply src/components/Sidebar/Sidebar.jsx]: Not through me. :male_detective: :laughing:

**Comment by montogeek** [comment]
Thanks

**Pull request merged**